src/daos-gen/daos-gen.c

```diff
@@ -0,0 +1,290 @@
+
+    int option_index = 0;
+    static struct option long_options[] = {
+        {"pool"                 , required_argument , 0, 'p'},
+        {"num-objects"          , required_argument , 0, 'o'},
+        {"keys-per-object"      , required_argument , 0, 'k'},
+        {"verbose"              , no_argument       , 0, 'v'},
+        {"quiet"                , no_argument       , 0, 'q'},
+        {"help"                 , no_argument       , 0, 'h'},
+        {0                      , 0                 , 0,  0 }
+    };
+
+    /* TODO: currently only generates DAOS_OF_KV_FLAT data, but could be
+     * updated to include more object types */
+
+    /* Parse options */
+    int num_objects = 0;
+    int keys_per_object = 0;
+
+    /* DAOS vars */ 
+    daos_args_t* daos_args = daos_args_new();    
+
+    int usage = 0;
+    while(1) {
+        int c = getopt_long(
+                    argc, argv, "p:o:k:vqh",
+                    long_options, &option_index
+                );
+
+        if (c == -1) {
+            break;
+        }
+
+        switch(c) {
+            case 'p':
+                uuid_parse(optarg, daos_args->src_pool_uuid);
+                break;
+            case 'o':
+                num_objects = atoi(optarg);
+                if (rank == 0) {
+                    MFU_LOG(MFU_LOG_INFO, "generating %d objects\n", num_objects);
+                }
+                break;
+            case 'k':
+                keys_per_object = atoi(optarg);
+                if (rank == 0) {
+                    MFU_LOG(MFU_LOG_INFO, "generating %d keys per object\n", keys_per_object);
+                }
+                break;
+            case 'v':
+                mfu_debug_level = MFU_LOG_VERBOSE;
+                break;
+            case 'q':
+                mfu_debug_level = MFU_LOG_NONE;
+                break;
+            case 'h':
+                usage = 1;
+                break;
+            case '?':
+                usage = 1;
+                break;
+            default:
+                if(rank == 0) {
+                    printf("?? getopt returned character code 0%o ??\n", c);
+                }
+        }
+    }
+
+    /* If we need to print the usage
+     * then do so before internal processing */
+    if (usage) {
+        if (rank == 0) {
+            print_usage();
+        }
+        mfu_finalize();
+        MPI_Finalize();
+        return 1;
+    }
+
+    int tmp_rc;
+
+    rc = daos_init();
+    if (rc != 0) {
+        MFU_LOG(MFU_LOG_ERR, "Failed to initialize daos");
+        mfu_finalize();
+        MPI_Finalize();
+        return 1;
+    }
+
+    /* set default number of objects and keys per object,
+     * if one is not passed in */
+    if (num_objects == 0)
+        num_objects = 10;
+    if (keys_per_object == 0)
+        keys_per_object = 20;
+
+    daos_obj_id_t       oid[num_objects];
+	daos_handle_t	    oh[num_objects];
+	char		        buf[BUFLEN];
+    const char          *key_fmt = "key%d";
```

Comment:
Have some tabs and spaces mixed in this part.
```suggestion
    daos_obj_id_t       oid[num_objects];
    daos_handle_t        oh[num_objects];
    char    	        buf[BUFLEN];
    const char          *key_fmt = "key%d";
```